src/main/java/pl/arturzgodka/databaseutils/CharacterDao.java

```diff
@@ -0,0 +1,150 @@
+package pl.arturzgodka.databaseutils;
+
+import jakarta.persistence.criteria.CriteriaBuilder;
+import jakarta.persistence.criteria.CriteriaQuery;
+import jakarta.persistence.criteria.Root;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import pl.arturzgodka.datamodel.*;
+
+import java.util.List;
+import java.util.Map;
+
+public class CharacterDao {
+
+    private final SessionFactory sessionFactory = UserSessionFactory.getCustomUserSessionFactory();
+    public void saveCharacter(CharacterDataModel characterDataModel) {
+        Session session = sessionFactory.openSession();
+        Transaction t = session.beginTransaction();
+        session.merge(characterDataModel);
+        t.commit();
+        session.close();
+    }
+
+    public void deleteCharacter(CharacterDataModel characterDataModel) {
+        Session session = sessionFactory.openSession();
+        Transaction transaction = session.beginTransaction();
+        session.remove(characterDataModel);
+        transaction.commit();
+        session.close();
+    }
+
+    public void deleteCharacter(int id) {
```

Comment:
why 2 methods doing the same?

Reply:
We need to discuss which fits better. To use as a parameter object or just id of object in data base.